### ISSUE TITLE
Validation not working when when and default statements are present together in Libyang1

### ISSUE THREAD

**Issue by azeem0396** [CESNET/libyang#1786]
Hi. I have the below rpc defined in my yang model.

```
    rpc example {
        description "Example rpc";
        input {
            leaf type {
                type enumeration {
                    enum op1 {
                        value 1;
                        description "op1";
                    }
                    enum op2 {
                        value 2;
                        description "op2";
                    }
                }
                mandatory true;
                description "Type";
            }
            leaf action {
                when "../type='op1'";
                type uint32;
                default 10;
                description "delete the partition.";
                }
            }
}
```

The RPC im sending is 

```
   const char* filter = <example xmlns=\"http://xyz.com/yang/xyz/rpc\"><type>op2</type></example>
   struct nc_rpc* rpc = nc_rpc_act_generic_xml(filter,NC_PARAMTYPE_CONST);
    msgtype = nc_send_rpc(sess, rpc, 1000, &msgid);
    if (msgtype == NC_MSG_ERROR)
    {
    printf("Failed to send the RPC\n");
    }
```
However the function fails as nc_send_rpc returns NC_MSG_ERROR.  The logs are

```
libyang[0]: When condition "../type = 'op2' not satisfied. (path: /xyz-rpc:example/action)
libyang[2]: Auto-deleting node "/xyz-rpc:example/action" due to when condition (../type = 'op1')
libyang[2]: All data nodes and constraints resolved.
Failed to send the RPC
```

I am able to send the rpc if I remove the default statement or if the type satisfies the when clause. Is there an issue with how libyang deals with when and default statements ?

We are currently using libyang.so.1.10.32

**Comment by michalvasko** [comment]
I think I have fixed the problem (commit 4575ef4ee942566246ae26713dd115dc5af75cf6). But the cause was some chaotic code, whose removal was one of the points of *libyang* v2 so you should migrate to avoid similar problem in future.

**Comment by azeem0396** [comment]
Yes it is working now. Thank you.
